### PR TITLE
Run critest on arm64

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Run critest on arm64 hardware.

We're also switching to dedicated cache usage to allow multi-arch caches.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@cri-o/cri-o-maintainers PTAL

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -14,37 +14,87 @@ permissions:
...
 jobs:
-  critest:
-    runs-on: ubuntu-latest
+  test-binaries:
+    strategy:
+      matrix:
+        run:
+          - runner: ubuntu-latest
+            arch: amd64
+          - runner: actuated-arm64-4cpu-16gb
```
Comment (alexellis) [inline .github/workflows/integration.yml:23]: Do you want to add in metering to see what kind of usage it tends to run up?

https://gist.github.com/alexellis/1f33e581c75e11e161fe613c46180771

Comment (saschagrunert) [reply .github/workflows/integration.yml:23]: Yep, done :+1:

**Pull request merged**